src/main/java/QueryServlet.java

```diff
@@ -50,45 +51,68 @@
               "female",
               "S0701_C01_013E,S0701_C04_013E"));
 
+  // After 2013, some queries have more data available
+  Map<String, Map<String, String>> queryToDataRowPost2013 =
+      ImmutableMap.of(
+          "live",
+          ImmutableMap.of(
+              "under-18",
+              "K200102_001E",
+              "all-ages",
+              "K200104_001E",
+              "male",
+              "K200101_002E",
+              "female",
+              "K200101_003E"),
+          "work",
+          ImmutableMap.of("all-ages", "K202301_004E", "over-18", "K202301_004E"),
+          "moved",
+          /* TODO: K200701_005E + K200701_006E for state query,
+          but actually have to add K200701_004E as well for county query */
+          ImmutableMap.of("all-ages", "S0201_119E,S0201_126E"));
+
   Map<String, String> tableNameToAbbrev =
       ImmutableMap.of("profile", "DP", "spp", "SPP", "subject", "ST");
 
   // Depending on the beginning of the data table string, the query URL changes slightly
   private String getDataTableString(String tablePrefix) {
-    if (tablePrefix.substring(0, 1).equals("D")) {
-      return "profile?get=NAME,";
+    String firstChar = tablePrefix.substring(0, 1);
+    if (firstChar.equals("K")) {
+      return "";
+    } else if (firstChar.equals("D")) {
+      return "/profile";
     } else if (tablePrefix.substring(0, 5).equals("S0201")) {
```

Comment:
Substring calls with endPrefix > length of string will cause an IndexOutOfBoundsException to be thrown. So we should be careful with these calls as it can cause the Servlet to crash.

Reply:
That's a good point. The input will definitely always have at least one character, but five is not as much of a guarantee so I will add an extra check there.